src/hotspot/share/opto/vectorIntrinsics.cpp

```diff
@@ -789,21 +789,28 @@ bool LibraryCallKit::inline_vector_shuffle_to_vector() {
 // <M,
 //  S extends VectorSpecies<E>,
 //  E>
-// M broadcastCoerced(Class<? extends M> vmClass, Class<E> elementType, int length,
-//                    long bits, S s,
+// M fromBitsCoerced(Class<? extends M> vmClass, Class<E> elementType, int length,
+//                    long bits, int mode, S s,
 //                    BroadcastOperation<M, E, S> defaultImpl)
-bool LibraryCallKit::inline_vector_broadcast_coerced() {
+bool LibraryCallKit::inline_vector_frombits_coerced() {
   const TypeInstPtr* vector_klass = gvn().type(argument(0))->isa_instptr();
   const TypeInstPtr* elem_klass   = gvn().type(argument(1))->isa_instptr();
   const TypeInt*     vlen         = gvn().type(argument(2))->isa_int();
+  Node* bits                      = argument(3);
+  // Mode argument determines the mode of operation it can take following values:-
+  // MODE_BROADCAST for vector Vector.boradcast and VectorMask.maskAll operations.
+  // MODE_BITS_COERCED_LONG_TO_MASK for VectorMask.fromLong operation.
+  const TypeInt*     mode         = gvn().type(argument(5))->isa_int();
 
-  if (vector_klass == NULL || elem_klass == NULL || vlen == NULL ||
-      vector_klass->const_oop() == NULL || elem_klass->const_oop() == NULL || !vlen->is_con()) {
+  if (vector_klass == NULL || elem_klass == NULL || vlen == NULL || mode == NULL ||
+      vector_klass->const_oop() == NULL || elem_klass->const_oop() == NULL ||
+      !vlen->is_con() || !mode->is_con()) {
```

Comment:
I think we need to check `bits` that it is `long` type node. It was not checked in old code at all.
And also add it to following print.